registration/api/v2/views.py

```diff
@@ -401,6 +401,22 @@ def static_content_overrides(self, request, pk=None):
 
             return Response(status=status.HTTP_200_OK, data=application.draft_static_content_overrides)
 
+    def list(self, request, *args, **kwargs):
+        """
+        Endpoint for getting the list of instances owned by the current user
+
+        If `fail_if_staff` is set for a staff user, the request returns a 400 error. This is used by the frontend to
+        prevent staff users from accidentally accessing and updating other users' instances.
+        """
+        if request.query_params.get("fail_if_staff", False) and request.user.is_staff:
```

Comment:
As mention in a comment above, we can probably remove the query parameter altogether and always fail with the error if user is staff.